integrations/common/marquez/provider/bigquery.py

```diff
@@ -0,0 +1,278 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import logging
+import traceback
+import attr
+
+from typing import Tuple, Optional, Dict, List
+
+from google.cloud import bigquery
+
+from marquez.dataset import Dataset, Source
+from marquez.models import DbTableSchema, DbColumn, DbTableName
+from marquez.schema import GITHUB_LOCATION
+from openlineage.facet import BaseFacet
+
+from marquez.utils import get_from_nullable_chain
+
+_BIGQUERY_CONN_URL = 'bigquery:{}'
+
+
+@attr.s
+class BigQueryErrorRunFacet(BaseFacet):
+    """
+    Represents errors that can happen during execution of BigqueryExtractor
+    :param clientError: represents errors originating in bigquery client
+    :param parserError: represents errors that happened during parsing SQL provided to bigquery
+    """
+    clientError: str = attr.ib(default=None)
+    parserError: str = attr.ib(default=None)
+
+    @staticmethod
+    def _get_schema() -> str:
+        return GITHUB_LOCATION + "bq-error-run-facet.json"
+
+
+@attr.s
+class BigQueryStatisticsRunFacet(BaseFacet):
+    """
+    Facet that represents relevant statistics of bigquery run.
+    :param cached: bigquery caches query results. Rest of the statistics will not be provided
+        for cached queries.
+    :param billedBytes: how many bytes bigquery bills for.
+    :param properties: full property tree of bigquery run.
+    """
+    cached: bool = attr.ib()
+    billedBytes: int = attr.ib(default=None)
+    properties: str = attr.ib(default=None)
+
+    @staticmethod
+    def _get_schema() -> str:
+        return GITHUB_LOCATION + "bq-statistics-run-facet.json"
+
+
+@attr.s
+class BigQueryStatisticsDatasetFacet(BaseFacet):
+    """
+    Facet that represents statistics of output dataset resulting from bigquery run.
+    :param outputRows: how many rows query produced.
+    :param size: size of output dataset in bytes.
+    """
+    rowCount: int = attr.ib()
+    size: int = attr.ib()
+
+    @staticmethod
+    def _get_schema() -> str:
+        return GITHUB_LOCATION + "bq-statistics-dataset-facet.json"
+
+
+@attr.s
+class BigQueryStatistics:
+    run_facets: Dict[str, BaseFacet] = attr.ib()
+    inputs: List[Dataset] = attr.ib()
+    output: Optional[Dataset] = attr.ib(default=None)
+
+
+class BigQueryStatisticsProvider:
```

Comment:
This class returns more than just stats, right? All input/output datasets and facets are returned here. Maybe `BigQueryDatasetsProvider`?